ignite/handlers/base_logger.py

```diff
@@ -161,11 +161,20 @@ def key_str_tf(tag: str, name: str, *args: str) -> str:
             elif isinstance(value, torch.Tensor) and value.ndimension() == 1:
                 for i, v in enumerate(value):
                     metrics_state_attrs_dict[key_tf(self.tag, name, str(i))] = v.item()
+            elif isinstance(value, Mapping):
+                for key, sub_value in value.items():
+                    if isinstance(value, numbers.Number):
+                        metrics_state_attrs_dict[key_tf(self.tag, name, key)] = value
+                    elif isinstance(value, torch.Tensor) and value.ndimension() == 0:
+                        metrics_state_attrs_dict[key_tf(self.tag, name, key)] = value.item()
+                    elif isinstance(value, torch.Tensor) and value.ndimension() == 1:
+                        for i, v in enumerate(value):
+                            metrics_state_attrs_dict[
+                                key_tf(self.tag, name, key, str(i))] = v.item()
+            elif isinstance(value, str) and log_text:
+                metrics_state_attrs_dict[key_tf(self.tag, name)] = value
```

Comment:
Yes, this makes sense to add the support, thanks!
I would implement that however a bit differently using a recursive function call, something like here: 
https://github.com/pytorch/ignite/blob/8782ca45204a083b3150271895751fd8fe0046d6/ignite/utils.py#L58-L80
This way we would avoid duplicating the code.

Reply:
Yes, I considered a form of recursive behavior, but I don't know how deep you would allow such feature? If there is no limit, no problem!

Reply:
Mainly, the idea is to remove duplicated code: 
```python
                    if isinstance(value, numbers.Number):
                        metrics_state_attrs_dict[key_tf(self.tag, name, key)] = value
                    elif isinstance(value, torch.Tensor) and value.ndimension() == 0:
                        metrics_state_attrs_dict[key_tf(self.tag, name, key)] = value.item()
                    elif isinstance(value, torch.Tensor) and value.ndimension() == 1:
                        for i, v in enumerate(value):
                            metrics_state_attrs_dict[
                                key_tf(self.tag, name, key, str(i))] = v.item()
```

Reply:
I'll look into that